interface/batchcom/smsnotification.php

```diff
@@ -28,72 +28,69 @@
     exit();
 }
 
- // default value
-$next_app_date = date("Y-m-d");
-$hour = "12";
-$min = "15";
-$provider_name = "EMR Group";
-$message = "Welcome to EMR Group";
-
 // process form
 if ($_POST['form_action'] == 'save') {
     if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
         CsrfUtils::csrfNotVerified();
     }
 
-    //validation uses the functions in notification.inc.php
-    //validate dates
-    if (!check_date_format($_POST['next_app_date'])) {
-        $form_err .= xl('Date format for "Next Appointment" is not valid') . '<br />';
+    if (! is_numeric($_POST['notification_id'])) {  // shouldn't happen
+        $form_err .= xl('Missing/invalid notification id') . '<br />';
     }
 
-    // validate selections
-    if ($_POST['sms_gateway_type'] == "") {
+    if (empty($_POST['sms_gateway_type'])) {
         $form_err .= xl('Error in "SMS Gateway" selection') . '<br />';
     }
 
-    // validates and or
-    if ($_POST['provider_name'] == "") {
+    if (empty($_POST['provider_name'])) {
         $form_err .= xl('Empty value in "Name of Provider"') . '<br />';
     }
 
-    if ($_POST['message'] == "") {
+    if (empty($_POST['message'])) {
         $form_err .= xl('Empty value in "SMS Text"') . '<br />';
     }
 
-    //process sql
+    // Store the new settings.  email_sender and email_subject are not used
+    // by SMS, but must be present because they are NOT NULL.  next_app_time,
+    // next_app_date, and notification_sent_date don't appear to be used by
+    // anyone.  notification_id is the pk, and should always be 1 for
+    // SMS settings (because that's how the db was seeded).
+
     if (!$form_err) {
-        $next_app_time = $_POST[hour] . ":" . $_POST['min'];
-        $sql_text = " ( `notification_id` , `sms_gateway_type` , `next_app_date` , `next_app_time` , `provider_name` , `message` , `email_sender` , `email_subject` , `type` ) ";
-        $sql_value = " (?, ?, ?, ?, ?, ?, ?, 'SMS') ";
-        $values = array($_POST['notification_id'], $_POST['sms_gateway_type'], $_POST['next_app_date'], $next_app_time,
-                        $_POST['provider_name'], $_POST['message'], $_POST['email_sender'], $_POST['email_subject']);
+        $sql_text = " ( `notification_id` , `sms_gateway_type` , `next_app_date` , `next_app_time` , `provider_name` , `message` , `email_sender` , `email_subject` , `type`, `notification_sent_date` ) ";
+        $sql_value = " (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ";
+        $values = array($_POST['notification_id'], $_POST['sms_gateway_type'],
+                        '0000-00-00', ':',
```

Comment:
same issue as above on the 0000-00-00 date

Reply:
Agree.  Happy to change it to NULL if you think it's advisable.